federation/pkg/federation-controller/replicaset/planner/planer_test.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package planer
```

Comment:
I would much prefer table driven testing here. 

e.g. http://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go

Could you rather have a table of tests (e.g. the 5 examples in your doc), and verify that the results are as intended?  That gives you better coverage now, and makes it easier for others to add more test cases later. 

Here's a simple example that I wrote recently:

https://github.com/kubernetes/kubernetes/blob/master/federation/pkg/dnsprovider/dns_test.go